R/MoBiProject.R

```diff
@@ -0,0 +1,170 @@
+    #' @param sourceFile (Optional) File used to load the project
+    #' @returns A new `MoBiProject` object.
+    initialize = function(netObject, sourceFile = NULL) {
+      super$initialize(netObject)
+      private$.sourceFile <- sourceFile
+    },
+
+    #' @description
+    #' Load a simulation from the project
+    #'
+    #' @param simulationName Name of the simulation.
+    #' @returns A `Simulation` object, if the simulation with the given name is present in the
+    #' project. `NULL` if no such simulation is available.
+    getSimulation = function(simulationName) {
+    },
+
+    #' @description
+    #' Get observed data present in the project.
+    #'
+    #' @param dataSetNames Optional. List of names of observed data sets to retrieve
+    #' from project. If `NULL`, all data sets are returned. If a specified data set
+    #' is not found, the name is ignored.
+    #'
+    #' @returns A named list of `DataSet` objects. `NULL` if the project does not contain
+    #' any observed data or no specified data sets are found.
+    getObservedData = function(dataSetNames = NULL){
+    },
+
+    #' @description
+    #' Get modules present in the project.
+    #'
+    #' @returns A named list of `MoBiModule` objects. `NULL` if the project does not contain
+    #' any module.
+    getModules = function(){
+    },
+
+    #' @description
+    #' Get a specified individual from the project
+    #' @param name Name of the individual
+    #' @param stopIfNotFound If `TRUE` (default), an error is thrown if the specified
+    #' individual is not present in the project.
+    #'
+    #' @returns An object of the type `BuildingBlock`. `NULL` if the project does not contain
+    #' such an individual and `stopIfNotFound = FALSE`.
+    getIndividual = function(name, stopIfNotFound = TRUE){
+    },
+    #' @description
+    #' Get specified expression profiles from the project.
+    #' @param names List of names of the expression profiles to retrieve.
+    #' @param stopIfNotFound If `TRUE` (default), an error is thrown if any of the specified
+    #' expression profiles is not present in the project.
+    #' @returns A named list of objects of the type `BuildingBlock`. `NULL` for each
+    #' specified expression profile that is not present in the project  if `stopIfNotFound = FALSE`.
+    getExpressionProfiles = function(names, stopIfNotFound = TRUE){
+    },
+
+    #' @description
+    #' Create a simulation configuration.
+    #'
+    #'
+    #' @param modulesNames A list of the modules from which to create in simulation.
+    #' All defined modules must be present in the project. The order of module names defines the order in which the modules will be combined to a simulation!
+    #' @param individualName Optional, name of the individual.
+    #' @param expressionProfilesNames Optional, list of expression profiles to apply to the simulation.
+    #' @param selectedInitialConditions By default, the first Initial Conditions
```

Comment:
@rwmcintosh @benjaperez1983 @msevestre 

Is it somehow possible to implement it this way? So that there is the default behavior, that the first available BB is selected, by also being able to select NO IC BB?

Reply:
And what is the current behavior?

Reply:
ah, by default we will pre-select an IC or PV if there is only one. That's Presentation logic, not Core logic.